tests/moduletool/test_install.py

```diff
@@ -941,3 +942,386 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
+        autostd=False,
+        python_package_sources=[local_module_package_index, index.url],
+        python_requires=[Requirement.parse(module.ModuleV2Source.get_package_name_for("parent_module"))],
+        install_project=True,
+    )
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "Successfully installed inmanta-module-child-module-3.3.3 inmanta-module-parent-module-1.2.3",
+    )
+
+
+def test_version_snapshot(local_module_package_index: str, snippetcompiler, caplog, modules_v2_dir, tmpdir):
+    """
+    Make sure the logs contain the correct version snapshot after each module installation.
+    """
+    caplog.set_level(logging.INFO)
+
+    index: PipIndex = PipIndex(artifact_dir=os.path.join(str(tmpdir), ".custom-index"))
+
+    # Prepare the modules:
+    # module a with version 1.0.0 and 5.0.0
+    # module b that depends on module a>=1.0.0
+    # module c that depends on module a==1.0.0
+
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "module_a"),
+        new_version=version.Version("1.0.0"),
+        new_name="module_a",
+        install=False,
+        publish_index=index,
+    )
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "module_a_hi"),
+        new_version=version.Version("5.0.0"),
+        new_name="module_a",
+        install=False,
+        publish_index=index,
+    )
+    module_b: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "module_b"),
+        new_version=version.Version("1.2.3"),
+        new_name="module_b",
+        install=False,
+        new_requirements=[InmantaModuleRequirement.parse("module_a>=1.0.0")],
+        publish_index=index,
+    )
+    module_c: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "module_c"),
+        new_version=version.Version("8.8.8"),
+        new_name="module_c",
+        install=False,
+        new_requirements=[InmantaModuleRequirement.parse("module_a==1.0.0")],
+        publish_index=index,
+    )
+
+    # Scenario 1
+    # Installing module b
+    snippetcompiler.setup_for_snippet(
+        f"""
+        import {module.ModuleV2.get_name_from_metadata(module_b)}
+        """,
+        autostd=False,
+        python_package_sources=[local_module_package_index, index.url],
+        python_requires=[Requirement.parse(module.ModuleV2Source.get_package_name_for("module_b"))],
+        install_project=True,
+    )
+
+    # We expect the latest version for a and b to be newly added
+    log_contains(
+        caplog,
+        "inmanta.module",
+        logging.INFO,
+        (
+            """
+Snapshot of modules versions post-install:\n
++inmanta-module-module-a: 5.0.0\n
++inmanta-module-module-b: 1.2.3"""
```

Comment:
Why is there line break character there? That means that there are two line breaks after eachother?